### PR TITLE
Fix pinpointer flicker

### LINKED ISSUE #15364: Pinpointer flickers

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

fixes #15364

plus i did some refactoring, deleted visualizer, used auto generated comp state and got rid of obsolete warnings, bla bla bla
**Media**

(gif may lag, but in game it looks just like before)
Peek 2023-04-16 13-30

**Changelog**

:cl:
- fix: fixed pinpointer flickering!
- fix: fixed pinpointer screen not being visible in dark!

### CODE REVIEW

File: Content.Client/Pinpointer/PinpointerVisualizerSystem.cs [hunk 1]
```diff
@@ -2,59 +2,58 @@
 using JetBrains.Annotations;
 using Robust.Client.GameObjects;
 
-namespace Content.Client.Pinpointer
+namespace Content.Client.Pinpointer;
+
+[UsedImplicitly]
+public sealed class PinpointerVisualizerSystem : VisualizerSystem<PinpointerComponent>
```
Comment (AJCM-git) [inline Content.Client/Pinpointer/PinpointerVisualizerSystem.cs:8]: just merge this with the clientside pinpointer system

File: Content.Client/Pinpointer/PinpointerVisualizerSystem.cs [hunk 2]
```diff
@@ -2,59 +2,58 @@
...
+                sprite.LayerSetState(PinpointerLayers.Screen, "pinonclose");
+                sprite.LayerSetRotation(PinpointerLayers.Screen, angle);
+                break;
+            case Distance.Medium:
+                sprite.LayerSetState(PinpointerLayers.Screen, "pinonmedium");
+                sprite.LayerSetRotation(PinpointerLayers.Screen, angle);
+                break;
+            case Distance.Far:
+                sprite.LayerSetState(PinpointerLayers.Screen, "pinonfar");
+                sprite.LayerSetRotation(PinpointerLayers.Screen, angle);
```
Comment (AJCM-git) [inline Content.Client/Pinpointer/PinpointerVisualizerSystem.cs:51]: male the hardcoded strings component fields

File: Content.Shared/Pinpointer/PinpointerComponent.cs [hunk 3]
```diff
@@ -1,57 +1,58 @@
...
-namespace Content.Shared.Pinpointer
+namespace Content.Shared.Pinpointer;
+
+/// <summary>
+/// Displays a sprite on the item that points towards the target component.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+[AutoGenerateComponentState]
+[Access(typeof(SharedPinpointerSystem))]
+public sealed partial class PinpointerComponent : Component
```
Comment (AJCM-git) [inline Content.Shared/Pinpointer/PinpointerComponent.cs:12]: make all fields VVable with readwrite where possible

**Review by AJCM-git — DISMISSED** [review]

**Comment by Slava0135** [comment]
animation is not playing right now - after I made it use GenericVisualizer

**Comment by Slava0135** [comment]
animation is working again, and it seems to be working correctly

File: Content.Client/Pinpointer/PinpointerSystem.cs [hunk 4]
```diff
@@ -1,78 +1,41 @@
...
+        // we need to update this arrow in a update loop
+        var query = EntityQueryEnumerator<PinpointerComponent, SpriteComponent>();
+        while (query.MoveNext(out var _, out var pinpointer, out var sprite))
         {
-            if (!Resolve(uid, ref pinpointer) || !pinpointer.HasTarget)
+            if (!pinpointer.HasTarget)
                 return;
-
             var eye = _eyeManager.CurrentEye;
             var angle = pinpointer.ArrowAngle + eye.Rotation;
```
Comment (metalgearsloth) [inline Content.Client/Pinpointer/PinpointerSystem.cs:26]: Won't this break with multi-viewport.

Comment (metalgearsloth) [reply Content.Client/Pinpointer/PinpointerSystem.cs]: e.g. client can have cameras open in a viewport which uses a different eye, so those pinpointers will look incorrect.

Comment (Slava0135) [reply Content.Client/Pinpointer/PinpointerSystem.cs]: well, i just used the old code, maybe fix it later

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**